code/datums/ammo/bullet/bullet.dm

```diff
@@ -53,10 +53,13 @@
 		fired_from.delete_bullet(firing_projectile, TRUE)
 		return
 
-	if(!(fired_from.flags_gun_features & GUN_SILENCED))
-		playsound(user, fired_from.fire_sound, fired_from.firesound_volume, FALSE)
-	else
+	if(fired_from.flags_gun_features & GUN_SILENCED)
 		playsound(user, fired_from.fire_sound, 25, FALSE)
+	else if(is_mainship_level(user.z) && !TIMER_COOLDOWN_CHECK(user, COOLDOWN_SHIPWIDE_EXECUTION_ALERT))
+		playsound_z(SSmapping.levels_by_any_trait(list(ZTRAIT_MARINE_MAIN_SHIP)), fired_from.fire_sound, volume = fired_from.firesound_volume)
```

Comment:
Do you reckon that you could play this as a normal sound but just with a large range and falloff so people can hear it across the other side of the Almayer but much quieter than next to it?

Reply:
I have no way to measure it, but if we wanted to make it shipwide, if someone gets executed in CIC (x 100, y 150) someone in the lower deck's reactor room (x 260 y 50) is incredibly far from it. Would it not be way more expensive to send an audio file that far vs using playsound_z?

Reply:
I don't know how it would compare resource wise, but for a thing that happens once in a blue moon it doesn't really matter. You could proccall playsound on a turf and then ghost to various places on the almayer then call it to judge the distance and falloff.

Reply:
let me know if the idea itself is approved, if so, I'll start messing around with the sound range and falloff